tests/README.md

```diff
@@ -0,0 +1,161 @@
+# Introduction
+
+This test suite is intended for whole program tests, i.e. for running a binary
```

Comment:
[suggestion] I think you can refer to these "whole program tests" as "**system tests**".

Reply:
I have used your wording, as I found my odd in the first place.